File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"
int steps (int start){
    int count = 0;
    if(start <= 0) return ERROR_VALUE;
    for(int n=start; n!=1; count++){
        if(n==1) return count;
        if(n % 2 == 0){    //Is even
            n /= 2;        //x/2
        }
        else n = 3*n+1;     //3x+1
    }
    return count;
}

File: solutions/zig/collatz-conjecture/1/collatz_conjecture.zig
pub const ComputationError = error{
    IllegalArgument,
};

pub fn steps(number: usize) anyerror!usize {
    if (number == 0) return ComputationError.IllegalArgument;
    var n = number;
    var result: usize = 0;
    while (n != 1) : (result += 1) {
        if (n & 1 == 1) {   // Odd
            n = (n * 3) + 1;
        } else {            // Even
            n /= 2;
        }
    }
    return result;
}
